app/Http/Controllers/LinkController.php

```diff
@@ -82,18 +83,19 @@ public function performRedirect(Request $request, $short_url, $secret_key=false)
         }
 
         // Increment click count
-        $long_url = $link->long_url;
-        $clicks = intval($link->clicks);
+        $referer = $request->server('HTTP_REFERER');
+        $ip = $request->getClientIp();
 
-        if (is_int($clicks)) {
-            $clicks += 1;
-        }
-        $link->clicks = $clicks;
-        $link->save();
+        Click::create([
```

Comment:
More advanced statistics should be an optional feature; we should still continue to increment clicks on the link model whether advanced stats are enabled or not. If advanced stats are disabled, we should avoid inserting rows for each click into the database and hide the advanced features. This will require a  toggle in the setup.